pkg/controller/model/grafanaServiceAccount.go

```diff
@@ -45,5 +53,6 @@ func GrafanaServiceAccountReconciled(cr *v1alpha1.Grafana, currentState *v1.Serv
 	reconciled := currentState.DeepCopy()
 	reconciled.Labels = getServiceAccountLabels(cr)
 	reconciled.Annotations = getServiceAccountAnnotations(cr, currentState.Annotations)
+	reconciled.ImagePullSecrets = getServiceAccountImagePullSecrets(cr)
```

Comment:
This triggers an infinite reconciliation when Kubernetes/OpenShift try to automatically assign a value for the image pull secret and the operator overwrites it on update. Had this issue when testing on OpenShift. I think on update we have to merge the existing image pull secrets with the ones provided in the CR. Similar to how annotations have to be merged to prevent overwriting auto assigned ones: https://github.com/integr8ly/grafana-operator/blob/master/pkg/controller/model/utils.go#L22

Reply:
did this happen on update or creation ? @pb82 ill look into this, thanks for the pointing out :+1:

Reply:
It happens immediately after create because OpenShift tried to add a pull secret which triggered the operator to override it and so on... Thank for looking into it!